pkg/app/api/grpcapi/web_api.go

```diff
@@ -168,6 +170,107 @@ func (a *WebAPI) ListEnvironments(ctx context.Context, req *webservice.ListEnvir
 	}, nil
 }
 
+func (a *WebAPI) EnableEnvironment(ctx context.Context, req *webservice.EnableEnvironmentRequest) (*webservice.EnableEnvironmentResponse, error) {
+	if err := a.updateEnvironmentEnable(ctx, req.EnvironmentId, true); err != nil {
+		return nil, err
+	}
+	return &webservice.EnableEnvironmentResponse{}, nil
+}
+
+func (a *WebAPI) DisableEnvironment(ctx context.Context, req *webservice.DisableEnvironmentRequest) (*webservice.DisableEnvironmentResponse, error) {
+	if err := a.updateEnvironmentEnable(ctx, req.EnvironmentId, false); err != nil {
+		return nil, err
+	}
+	return &webservice.DisableEnvironmentResponse{}, nil
+}
+
+func (a *WebAPI) DeleteEnvironment(ctx context.Context, req *webservice.DeleteEnvironmentRequest) (*webservice.DeleteEnvironmentResponse, error) {
+	claims, err := rpcauth.ExtractClaims(ctx)
+	if err != nil {
+		a.logger.Error("failed to authenticate the current user", zap.Error(err))
+		return nil, err
+	}
+
+	if err := a.validateEnvBelongsToProject(ctx, req.EnvironmentId, claims.Role.ProjectId); err != nil {
+		return nil, err
+	}
+
+	if err := a.environmentStore.DeleteEnvironment(ctx, req.EnvironmentId); err != nil {
+		switch err {
+		case datastore.ErrNotFound:
+			return nil, status.Error(codes.NotFound, "The environment is not found")
+		case datastore.ErrInvalidArgument:
+			return nil, status.Error(codes.InvalidArgument, "Invalid value to delete")
+		default:
+			a.logger.Error("failed to delete the environment",
+				zap.String("env-id", req.EnvironmentId),
+				zap.Error(err),
+			)
+			return nil, status.Error(codes.Internal, "Failed to delete the environment")
+		}
+	}
+
+	return &webservice.DeleteEnvironmentResponse{}, nil
+}
+
+func (a *WebAPI) updateEnvironmentEnable(ctx context.Context, envID string, enable bool) error {
+	claims, err := rpcauth.ExtractClaims(ctx)
+	if err != nil {
+		a.logger.Error("failed to authenticate the current user", zap.Error(err))
+		return err
+	}
+
+	if err := a.validateEnvBelongsToProject(ctx, envID, claims.Role.ProjectId); err != nil {
+		return err
+	}
+
+	var updater func(context.Context, string) error
+	if enable {
+		updater = a.environmentStore.EnableEnvironment
+	} else {
+		updater = a.environmentStore.DisableEnvironment
+	}
+
+	if err := updater(ctx, envID); err != nil {
+		switch err {
+		case datastore.ErrNotFound:
+			return status.Error(codes.InvalidArgument, "The environment is not found")
```

Comment:
should be `codes.NotFound` 👀

Reply:
It should. All parts including Applications should get so.

Reply:
Addressed 651e3a8